### DISCUSSION TITLE
Advanced Search doesn't have keywords

### DISCUSSION THREAD

**Discussion by Skreeauk** [fuma-nama/fumadocs#581] [Q&A] [answered]
I looked through the code for the simple and advanced search (initSearchAPI, initSearchAPIAdvanced), and noticed that the simple search tracks keywords (not mentioned in the [docs](https://fumadocs.vercel.app/docs/headless/search/flexsearch)) while the advanced drops the keywords and now tracks structuredData (headings + content).

Is it possible for the advanced search to also track the keywords alongside structuredData?

**Comment by fuma-nama** [comment]
You can modify the structured data to include custom keywords. The keywords feature is undocumented (for historical reasons), and not recommended to use.

You can also use the tag feature, it was used to filter indexes by a tag, useful for implementing multiple page trees. I planned to document it after making some improvements on other features

**Reply by Skreeauk** [reply]
I'll see what I can do with the structuredData. From a glance, I don't think it will actually work as the advanced search still only parses headings and contents from structuredData.

For context on my side:
I'm trying to make some kind of wiki/guide for a game. There are official in game terms and nicknames from the community.
I have already added "keywords" to the frontmatter by extending the defaultSchema, so now in my mdx files, i can add keywords: ["stringA","stringB"].
I would then just plug in the keywords into the built-in simple search and it worked, however I also wanted the heading search as well from advanced.
Tag would work for 1 keyword, but not for multiple keywords

Ultimately, I would like to avoid making my own search function. Thanks for the quick response.

**Reply by fuma-nama** [reply, marked as answer]
I see, in this case we don't have a way to add keywords. I'll support that later, for now you can add the keywords as headings (in structured data) so that they can be searchable by users

**Reply by Skreeauk** [reply]
Thats what I ended up doing, thanks for taking it into consideration for a future feature.

I really liked fumadocs so far as I went from contentlayer into this. The approach of removing header also makes it really clean.

Thanks again for the quick response.
